test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionProtocolTests.cs

```diff
@@ -320,5 +324,57 @@ public class HubConnectionProtocolTests
                 await connection.DisposeAsync().OrTimeout();
             }
         }
+
+        [Fact]
+        public async Task MessagesEncodedWhenUsingBinaryProtocolOverTextTransport()
+        {
+            var connection = new TestConnection(TransferMode.Text);
+
+            var hubConnection = new HubConnection(connection, new MessagePackHubProtocol(), new LoggerFactory());
+            try
+            {
+                await hubConnection.StartAsync().OrTimeout();
+                await hubConnection.SendAsync("MyMethod", 42).OrTimeout();
+
+                await connection.ReadSentTextMessageAsync().OrTimeout();
+                var invokeMessage = await connection.ReadSentTextMessageAsync().OrTimeout();
+
+                // this throws if the message is not a valid base64 string
+                Convert.FromBase64String(invokeMessage);
```

Comment:
I could add a separate test but it is not our code that throws but Convert itself so not sure what the value would be. The comment was supposed to only explain why this test does not have any `Assert`s